tutorials/interactive/index.md

```diff
@@ -0,0 +1,781 @@
+  collected by the garbage collector (GC).
+{% endcomment %}
+* All widgets are _immutable_, meaning once created they cannot change.
+* When a widget's appearance changes, a new widget is created to reflect the change,
+  but the State object allows the widget's state to persist.
+* Typically, the State object is managed by the widget itself, the parent widget,
+  or some combination.
+* The widget's implementation determines who manages the state.
+
+</div>
+
+There are two kinds of widgets, stateful and stateless.
+
+A _stateless_ widget
+is drawn to the screen and sits there, informing the user,
+or making the UI look more attractive.
+The user doesn't interact with a stateless widget.
+[Icon,](https://docs.flutter.io/flutter/material/Icon-class.html)
+[Image,](https://docs.flutter.io/flutter/widgets/Image-class.html) and
+[Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are
+examples of stateless widgets, which subclass
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic. The user can interact with it
+(by moving a slider, or checking a box, for example),
+or it changes over time (the UI updates as a result of a data feed, for example).
+[Checkbox,](https://docs.flutter.io/flutter/material/Checkbox-class.html)
+[Radio,](https://docs.flutter.io/flutter/material/Radio-class.html) and
+[Slider](https://docs.flutter.io/flutter/material/Slider-class.html) are
+examples of stateful widgets, which subclass
+[StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html).
+
+<a name="stateful-widget-pattern-1"></a>
+## Creating a stateful widget
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* To create a custom stateful widget, subclass StatefulWidget and State.
+* The State object manages the widget's state.
+* Calling `setState()` updates the widget's UI.
+
+</div>
+
+In this section, you'll create a custom stateful widget.
+You'll be replacing two stateless widgets,
+the solid red star and the numeric count next to it, with a single custom
+stateful widget that manages a row with two children widgets: an IconButton
+and Text.
+
+A custom stateful widget requires creating two classes:
+
+1. A subclass of StatefulWidget that defines the widgets and describes how to
+   draw them in its `build()` method.
+2. A subclass of State that manages state for that widget.
+
+This section shows you how to build a stateful widget, called FavoriteWidget,
+for the Lakes app. The first step is choosing who manages the State object.
+
+<a name="step-1"></a>
+### Step 1: Decide who manages the widget's state
+
+There are different approaches to managing a widget's state, but in our example
+the widget itself, FavoriteWidget, will manage its own state.
```

Comment:
Technically, the State object contains the state. Not sure it's totally accurate to say "FavoriteWidget manages its own state", unless you mean an instance of State.